cmd/simulator/txs/agent.go

```diff
@@ -75,12 +76,6 @@ func (a issueNAgent[T]) Execute(ctx context.Context) error {
 		totalConfirmedTime time.Duration
 	)
 
-	defer func() {
-		if err := a.worker.Close(ctx); err != nil {
-			log.Error("error trying to close worker: %w", "err", err)
-		}
-	}()
-
```

Comment:
Removed so clients can be re-used. Leave closing clients as an option to the caller.